### DISCUSSION TITLE
Termination Reason MAX_EVENTS

### DISCUSSION THREAD

**Discussion by daniel-dierich** [camaraproject/SimSwap#176] [Q&A] [answered]
**Problem description**
Hi all, 

currently there is a Termination Reason `MAX_EVENTS_REACHED` defined. I know this max event logic from the other APIs. However there is no option to define this value. For example in Device Location geofencing subscriptions you can define the value  using `"subscriptionMaxEvents": 5` in your request.

**Expected behavior**
I would propose to add the request parameter.

**Alternative solution**
Alternatively the Termination Reason could be removed.

**Comment by gregory1g** [comment, marked as answer]
as far as I remember, we did not found a use case for subscripting to a fixed number of simswap events. 
At the same time the implementation would  require a state (the counter) to be kept (this would impact horizontal scalability of the service). Therefore the decision was - do no support even counter based subscription, while termination reason was kept to keep alignment with the general commonalities design.

**Comment by daniel-dierich** [comment]
I understand. Thank you for explaining!

**Comment by bigludo7** [comment]
@gregory1g @daniel-dierich I convert this issue in discussion as it did not required code change.
